core/src/main/java/org/infinispan/manager/DefaultCacheManager.java

```diff
@@ -678,7 +679,13 @@ public void start() {
    private void terminate(String cacheName) {
       CompletableFuture<Cache<?, ?>> cacheFuture = this.caches.get(cacheName);
       if (cacheFuture != null) {
```

Comment:
Shouldn't we also remove caches from wiredCaches when they are being removed ?

Reply:
The caches are not removed AFAIK. I was about to ask if this isn't a memory leak :)

Reply:
@pruivo I think @rvansa was saying that we're not removing the caches from the `caches` map either, so one extra reference won't hurt. I have no idea what's happening now if I call `removeCache("x"); getCache("x")`, but I'd modify `removeCache()` to remove the cache from both maps.